modules/gapi/include/opencv2/gapi/s11n.hpp

```diff
@@ -300,14 +306,48 @@ static cv::util::optional<GCompileArg> exec(const std::string& tag, cv::gapi::s1
 
 template<typename T> struct deserialize_runarg;
 
-template<typename RMatAdapterType>
-struct deserialize_runarg {
+template<typename T, typename... Types> inline
+typename std::enable_if<std::is_base_of<RMat::Adapter, T>::value, GRunArg>::
+type deserializeRMatHelper(cv::gapi::s11n::IIStream& is, uint32_t) {
+    auto ptr = std::make_shared<T>();
+    ptr->deserialize(is);
+    return GRunArg { RMat(std::move(ptr)) };
+}
+
+template<typename T, typename... Types> inline
+typename std::enable_if<std::is_base_of<MediaFrame::IAdapter, T>::value, GRunArg>::
+type deserializeRMatHelper(cv::gapi::s11n::IIStream& is, uint32_t idx) {
```

Comment:
is it still `RMatHelper`? I see this case applies to `T` being a `MediaFrame::IAdapter`.

Also just noticed `RMat::Adapter` isn't `I`? I will file a task for 5.0 :)

Reply:
I added a note about how it works. Basically, yes - it's `RMatHelper` since it's being called for `RMat` `RunArg`, but we need `enable_if`s for all adapter bases (it won't compile the other way)